### DISCUSSION TITLE
On which macOS versions is the use_bdaddr argument working?

### DISCUSSION THREAD

**Discussion by koenvervloesem** [hbldh/bleak#1260] [Q&A] [answered]
The 0.20.0 release notes show:

> Added optional hack to use Bluetooth address instead of UUID on macOS. Merged https://github.com/hbldh/bleak/pull/1073.

However, when trying to use the `use_bdaddr` argument, for instance with the `detection_callback.py` example, I still don't see real Bluetooth addresses. I tried this on macOS 13.0.1 and 13.2.1. In which versions of the OS is this known working?

**Comment by dlech** [comment, marked as answer]
python examples/detection_callback.py --macos-use-bdaddr

works for me

    ...
    2023-03-19 13:30:03,049 __main__ INFO: 01:D9:ED:A3:E9:F1: AdvertisementData(local_name='prime_hub', service_data={'00002a50-0000-1000-8000-00805f9b34fb': b'\x01\x97\x03\x81\x00\x00\x00'}, service_uuids=['c5f50001-8280-46da-89f4-6d8051e4aeef'], tx_power=0, rssi=-43)
    ...

**Reply by koenvervloesem** [reply]
Ok, it seems I botched my Python environment during the OS upgrades, it works now after a clean install of the packages, thanks for the confirmation!
